scapy/sendrecv.py

```diff
@@ -887,7 +887,8 @@ def _write_to_pcap(packets_list):
 
                 if isinstance(offline, Packet):
                     tempfile_written, offline = _write_to_pcap([offline])
-                elif isinstance(offline, list) and \
+                elif (isinstance(offline, list) or
+                        isinstance(offline, PacketList)) and \
                         all(isinstance(elt, Packet) for elt in offline):
```

Comment:
We need to find something more efficient than this. We can probably just check the first element and assume the type list is consistent..

Reply:
Howabout: `isinstance(offline, (list, PacketList))` ?

Reply:
I think @gpotter2 was actually talking about the next check (`all(...)`)

Reply:
I agree with you @gpotter2 but that's not related to this PR. The proposed patch looks good to me.

Reply:
Ah ok thanks. Agree all() check can be a bit of a delay - Looking at where the call graph goes it ends in calling [PcapWriter()](https://github.com/secdev/scapy/blob/ef4937f3c907749a9b8689706070b35a819722d6/scapy/utils.py#L1754) whilst it checks for None type one could just add a `Packet` type check there...